src/ica_api.c

```diff
@@ -4017,14 +4037,38 @@ int ica_aes_gcm_kma_verify_tag(const unsigned char* known_tag, unsigned int tag_
 #endif /* NO_CPACF */
 }
 
+int ica_aes_gcm_kma_get_iv(const kma_ctx* ctx, unsigned char *iv, unsigned int *iv_length)
+{
+	if (ctx == NULL)
+		return EINVAL;
+
+	if (iv == NULL) {
+		*iv_length = ctx->iv_length;
+		return 0;
+	}
+
+	if (*iv_length < ctx->iv_length)
+		return ENOMEM;
```

Comment:
ENOMEM? Shouldn't this be EINVAL? ENOMEM is only when allocating memory failed as far as I know.

Reply:
Hmm .... well, the ENOMEM here shall indicate that the application did not provide a "big-enough" buffer to receive the internally created iv. Which is indeed kind of a "space" or "memory" problem. I can change this to EINVAL if this is the agreed rc for this kind of error.

Reply:
Update: just had a look at the libzpc: there we use an own rc ZPC_ERROR_SMALLOUTBUF to indicate exactly this case where the application provides not enough space. So no conflict there.
When looking at the definition of ENOMEM:
`#define ENOMEM		12	/* Not enough space */`
it just says "not enough space". Hmmm ...

Reply:
Well https://man7.org/linux/man-pages/man3/errno.3.html says "ENOMEM Not enough space/cannot allocate memory (POSIX.1-2001)."
https://www.gnu.org/software/libc/manual/html_node/Error-Codes.html says "ENOMEM
“Cannot allocate memory.” The system cannot allocate more virtual memory because its capacity is full."

Reply:
hm, ok, then I will change to EINVAL at both places ...